docs/buildrun.md

```diff
@@ -0,0 +1,123 @@
+- A referenced `Build` instance to use during the build construction.
+- Compute resources.
+- A service account for hosting all related secrets in order to build the image.
+
+A `BuildRun` is available within a namespace.
+
+## BuildRun Controller
+
+The controller watches for:
+
+- Updates on a `Build` resource (_CRD instance_)
+- Updates on a `TaskRun` resource (_CRD instance_)
+
+When the controller reconciles it:
+
+- Looks for any existing owned `TaskRuns` and update its parent `BuildRun` status.
+- Retrieves the specified `SA` and sets this with the specify output secret on the `Build` resource.
+- Generates a new tekton `TaskRun` if it does not exist, and set a reference to this resource(_as a child of the controller_).
+- One any subsequent updates on the `TaskRun`, the parent `BuildRun` resource instance will be updated.
+
+## Configuring a BuildRun
+
+The `BuildRun` definition supports the following fields:
+
+- Required:
+  - [`apiVersion`](https://kubernetes.io/docs/concepts/overview/working-with-objects/kubernetes-objects/#required-fields) - Specifies the API version, for example `build.dev/v1alpha1`.
+  - [`kind`](https://kubernetes.io/docs/concepts/overview/working-with-objects/kubernetes-objects/#required-fields) - Specifies the Kind type, for example `BuildRun`.
+  - [`metadata`](https://kubernetes.io/docs/concepts/overview/working-with-objects/kubernetes-objects/#required-fields) - Metadata that identify the CRD instance, for example the name of the `BuildRun`.
+  - `spec.buildRef` - Specifies an existing `Build` resource instance to use.
+
+- Optional:
+  - `spec.resources` - Refers to the compute [resources](https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/) used on the container where the image is build.
+  - `spec.serviceAccount` - Refers to the SA to use when building the image. (_defaults to the `default` SA_)
+
+### Defining the BuildRef
+
+A `BuildRun` resource can reference a `Build` resource, that indicates what image to build. For example:
+
+```yaml
+apiVersion: build.dev/v1alpha1
+kind: BuildRun
+metadata:
+  name: buildpack-nodejs-buildrun-namespaced
+spec:
+  buildRef:
+    name: buildpack-nodejs-build-namespaced
+```
+
+### Defining Resources
+
+A `BuildRun` resource can define resources, like **limits** to use in the pod where the build execution will take place, for example:
+
+```yaml
+apiVersion: build.dev/v1alpha1
+kind: BuildRun
+metadata:
+  name: kaniko-golang-buildrun
+spec:
+  buildRef:
+    name: kaniko-golang-build
+  resources:
+    limits:
+      cpu: "1"
+```
+
+### Defining the ServiceAccount
+
+A `BuildRun` resource can define a serviceaccount to use. Usually this SA will host all related secrets referenced on the `Build` resource, for example:
+
+```yaml
+apiVersion: build.dev/v1alpha1
+kind: BuildRun
+metadata:
+  name: buildpack-nodejs-buildrun-namespaced
+spec:
+  buildRef:
+    name: buildpack-nodejs-build-namespaced
+  serviceAccount: pipeline
```

Comment:
This is outdated. The `serviceAccount` property is now an object with a `generate` property.

Reply:
yes, I added a comment now:
```
You can also use set the `spec.serviceAccount.generate` path to `true`. This will generate the service account during runtime for you.
```

Reply:
@qu1queee further clarifying:

```yaml
serviceAccount: pipeline
```

is incorrect. It needs to be:

```yaml
serviceAccount:
  name: pipeline
```

See https://github.com/redhat-developer/build/blob/982e4b9519b81c7b5a45ee91e6f00824e3d11a7f/pkg/apis/build/v1alpha1/buildrun_types.go#L29 and https://github.com/redhat-developer/build/blob/982e4b9519b81c7b5a45ee91e6f00824e3d11a7f/pkg/apis/build/v1alpha1/buildrun_types.go#L66.

Reply:
yep, fixed, thanks